File: python/datashaper/datashaper/engine/verbs/strings/upper.py
#
# Copyright (c) Microsoft. All rights reserved.
# Licensed under the MIT license. See LICENSE file in the project.
#
"""Upper verb implementation."""
from typing import cast

import pandas as pd

from datashaper.engine.verbs.verb_input import VerbInput
from datashaper.engine.verbs.verbs_mapping import verb
from datashaper.table_store.types import VerbResult, create_verb_result


@verb(name="strings.upper")
def upper(input: VerbInput, column: str, to: str, **_kwargs: dict) -> VerbResult:
    """Upper verb implementation."""
    input_table = input.get_input()
    output = cast(pd.DataFrame, input_table)
    output[to] = output[column].str.upper()
    return create_verb_result(output)

File: javascript/workflow/src/verbs/strings/upper.ts
/*!
 * Copyright (c) Microsoft. All rights reserved.
 * Licensed under the MIT license. See LICENSE file in the project.
 */

import type { StringsArgs } from '@datashaper/schema'

import { op } from 'arquero'
import type { ColumnTableStep } from '../util/factories.js'
import { stepVerbFactory } from '../util/factories.js'

export const upperStep: ColumnTableStep<StringsArgs> = (
	input,
	{ column, to },
) => {
	return input
		.params({ column })
		.derive({ [to]: (d: any, $: any) => op.upper(d[$.column]) })
}

export const upper = stepVerbFactory(upperStep)
